### PR TITLE
Basic Authentication documentation

### PR DESCRIPTION
**Pull request by wild-endeavor**
Adding basic documentation for authentication and cleaning a couple things up.

### CODE REVIEW

File: rsts/administrator/install/authentication.rst [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+The gRPC client provided by the Flyte IDL, or direct calls to the HTTP endpoints on Admin from within a running script are ways that we have seen users hit the control plane directly.  We generally discourage this behavior as it leads to a possible self-imposed DOS vector, as they are generally made from within a running workflow itself. For instance, a Flyte task can fetch the definition for a launch plan associated with a completely different workflow, and then launch an execution of it. This is not the correct way to launch one workflow from another but for the time being remains possible.
+
+
+******
+Flow
+******
+
+Flyte Admin authentication is implemented using the authorization code flow.
+
+Flyte UI Flow
```
Comment (kumare3) [inline rsts/administrator/install/authentication.rst]: Can we move these images into our gh-pages branch

Comment (wild-endeavor) [reply rsts/administrator/install/authentication.rst]: Yes, I will do that.

File: rsts/administrator/install/authentication.rst [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+
+Flyte CLI Flow
+==============
+https://swimlanes.io/d/q64OxuoxT
+
+.. image:: flyte_cli_flow.png
+  :width: 600
+  :alt: Flyte CLI Swimlane
+
```
Comment (katrogan) [inline rsts/administrator/install/authentication.rst]: What do you think about adding to this section to go over the required env variables that need to be set in flytekit?

Comment (wild-endeavor) [reply rsts/administrator/install/authentication.rst]: move them up from down below?  I can.  Or I can just add a quick note to read the flyte-cli docs.  There's two parts of flytekit right?  I don't actually have a swimlane for the basic auth... i guess i can make one.

Comment (katrogan) [reply rsts/administrator/install/authentication.rst]: meh this is already great! i guess i got confused reading the CI part and thought it was for basic auth only - feel free to ignore the above coment

**Comment by wild-endeavor** [comment]
End to end tests are failing for this PR, but this is a broader issue.  A test PR was opened up on master with no changes tests there also failed.  End to end tests off of this branch are also passing when running locally, making it hard to understand what's happening.  The usual log output that we depend on to debug is also not getting printed by the GitHub workflow.

**Pull request merged**